.babelrc

```diff
@@ -0,0 +1,8 @@
+{
+    "presets": [
+        ["@babel/preset-env", {
+            "useBuiltIns": "entry",
+            "corejs": 3
+        }]
```

Comment:
@bakshiutkarsha Looking at the [Babel usage guide](https://babeljs.io/docs/en/usage), I think the issue with missing polyfills might be because you haven't specified any browsers in this setup file. E.g. the setup has these entries (NB don't use these, they need to be modified for our needs):
```
  "presets": [
    [
      "@babel/env",
      {
        "targets": {
          "edge": "17",
          "firefox": "60",
          "chrome": "67",
          "safari": "11.1",
        },
        "useBuiltIns": "usage",
        "corejs": "3.6.4",
      }
    ]
  ]
```
As a minimum we need to support:
```
"edge": "15",
"firefox": "52",
"ie": "11"
```
I'm not sure what we support for Chrome (e.g. minimum Android version) and Safari for iOS app), but with IE11, most other things are probably covered.

Reply:
@Jaifroid  I have done that in package.json [here](https://github.com/openzim/mwoffliner/pull/1237/files#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R145)

Reply:
Hmm. So any thoughts on why it's not working?

Two random guesses:

1. (I don't really know babel config, so correct me if I'm wrong): maybe adding the browser targets in `package.json` only affects the environment in which **mwoffliner** runs, i.e. `node.js` (I assume)? Whereas we need a config that will load corejs polyfills according to the environment in which `-/_api/toc.js` is running, i.e. the detection logic needs to run inside toc.js. That may be completely wrong...
2. You're using corejs 3 with `babel-preset-env`. I wonder if the breaking change mentioned in https://github.com/babel/babel/pull/7646 affects this setup:

> Updating @babel/polyfill to core-js@3 will break @babel/preset-env. @babel/preset-env with useBuiltIns transforms import of @babel/polyfill to import of required core-js modules. Most users don't add core-js dependency directly. In case of a minor update, even if @babel/polyfill dependency will be updated with @babel/preset-env, in the top level of node_modules we will have core-js@2, not core-js@3 because a serious part of dependencies depends on packages with core-js@2 in dependencies.

A not great alternative would be to provide our own closest function if `Element.closest` is undefined:
```
        // Create a closest function alternative because IE11 and others do not support closest
        function closest (ele, s) {
            var cele = ele;
            var cmatches = Element.prototype.matches || Element.prototype.msMatchesSelector || Element.prototype.webkitMatchesSelector;
            do {
                if (cmatches.call(cele, s)) return cele;
                cele = cele.parentElement || cele.parentNode;
            } while (cele !== null && cele.nodeType === 1);
            return null;
        };
```

Reply:
@bakshiutkarsha Can you help here or have you left this PR to someone else?

Reply:
Yeah, I can close this. Will implement the changes suggested by @Jaifroid .